RevenueCatUI/CustomerCenter/Views/RestorePurchasesAlert.swift

```diff
@@ -166,32 +153,27 @@ struct RestorePurchasesAlert: ViewModifier {
                 message += "\n\n" + localization[.updateWarningDescription]
             }
             return message
-        case .restorePurchases:
-            return localization[.goingToCheckPurchases]
         }
     }
 
     private func dismissAlert() {
-        self.alertType = .restorePurchases
         self.isPresented = false
+        Task {
+            await self.customerCenterViewModel.loadScreen()
+        }
```

Comment:
I'm not 100% sure about this, but I _think_ that it might happen that because the `RestorePurchasesAlert` view gets dismissed with `self.isPresented = false`, the `Task` created here, which inherits the context from the parent, could get canceled on dismissal. If we wanted to create an independent task that gets executed regardless, perhaps we should do a `Task.detached`, like this
```suggestion
        Task.detached {
            await self.customerCenterViewModel.loadScreen()
        }
        self.isPresented = false
```
With detached, we guarantee that it does not get canceled even if the parent context gets canceled due to the dismissal.

Reply:
@vegaro anyway, I think that a method in `CustomerCenterViewModel` to notify it of `onDismissRestorePurchasesAlert()` is a cleaner approach, since the `RestorePurchasesAlert` already has a reference to the view model, I think it makes total sense.
```suggestion
        self.onDismissRestorePurchasesAlert()
        self.isPresented = false
```
In the view model's `onDismissRestorePurchasesAlert()` method, a `Task` would be initiated. This way, we have the `Task` inherit the context of the view model and it's cleaner on the alert's side.

Reply:
right, I like that! Take a look at [8cc6d9b](https://github.com/RevenueCat/purchases-ios/pull/4930/commits/8cc6d9b0f6cb70a792c1ba49bbfa22cdd44e9542)

Reply:
Nice! I like this much better 👍